samcli/lib/build/app_builder.py

```diff
@@ -520,21 +519,28 @@ def _build_function(  # pylint: disable=R1710
 
                 options = ApplicationBuilder._get_build_options(function_name, config.language, handler)
                 # By default prefer to build in-process for speed
-                build_method = self._build_function_in_process
                 if self._container_manager:
-                    build_method = self._build_function_on_container
-                    return build_method(
+                    image = None
+                    if self._build_images is not None:
+                        if function_name in self._build_images:
+                            image = self._build_images[function_name]
+                        elif None in self._build_images:
```

Comment:
Can we give this a name or something and specify in the doc? I don't really support having None as a key

Reply:
We can but I don't see a reason to not use `None` as key for this use case. Having a special string for default would mean we need to make sure all references are matching with an extra constant.